testsuite/classlibrary/TestBuffer.sc

```diff
@@ -245,4 +245,36 @@ TestBuffer : UnitTest {
 		buffer.free;
 	}
 
+	test_getToFloatArray {
+		var server, data, condition, inTime = false;
+		condition = Condition.new;
+		this.bootServer;
+		server = Server.default;
+		data = [1.0, 5.0, 3.0, [9.0, -9.0], 2.0];
+
+		{
+			Duty.ar(SampleDur.ir, 0, Dseq(data, inf))
+		}.asBuffer(0.2, server, { |buffer|
+
+			buffer.getToFloatArray(0, data.size * 2, action: { |array|
+				condition.unhang;
+				this.assertEquals(array.as(Array), data.flop.lace,
+					"getToFloatArray should get correct data"
+				);
+				inTime = true;
+			})
+		});
+
+		fork {
+			3.wait;
+			condition.unhang;
```

Comment:
this is going to restart the thread if it's waiting somewhere else later; is each test for sure being run on a separate thread now once we add isolation?

Reply:
no, all tests are run in one single thread, they come one by one in order. Good you spotted this - I'll fix this (and another good reason we need `hangWithTimeout`)